src/plugins/kdb/ldap/libkdb_ldap/ldap_realm.c

```diff
@@ -275,8 +275,10 @@ krb5_ldap_delete_realm (krb5_context context, char *lrealm)
                     krb5_parse_name(context, values[i], &principal);
                     if (principal_in_realm_2(principal, lrealm) == 0) {
                         st=krb5_ldap_delete_principal(context, principal);
-                        if (st && st != KRB5_KDB_NOENTRY)
+                        if (st && st != KRB5_KDB_NOENTRY) {
+                            krb5_free_principal(context, principal);
                             goto cleanup;
+                        }
```

Comment:
`values` also leaks here.  We try to avoid repeating free statements in error handling blocks;  I will take a closer took at this function.